chromium_src/components/payments/content/payment_request.cc

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/payments/content/payment_request.h"
+
+#include "base/bind.h"
+#include "brave/components/brave_rewards/common/constants.h"
+#include "brave/components/payments/content/buildflags/buildflags.h"
+#include "content/public/browser/browser_context.h"
+#include "third_party/blink/public/mojom/payments/payment_request.mojom.h"
+
+using payments::mojom::PaymentErrorReason;
+
+#if BUILDFLAG(ENABLE_PAY_WITH_BAT)
+namespace payments {
+
+void PaymentRequest::OnError(PaymentErrorReason reason,
+                             std::string err) {
+  client_->OnError(reason, err);
+}
+
+void PaymentRequest::Init(
+    mojo::PendingRemote<mojom::PaymentRequestClient> client,
+    std::vector<mojom::PaymentMethodDataPtr> method_data,
+    mojom::PaymentDetailsPtr details,
+    mojom::PaymentOptionsPtr options) {
+  bool result = false;
```

Comment:
you're skipping all kinds of checks in `PaymentRequest::Init` like `is_initialized_` and other checks to make sure this is a secure origin